### PR TITLE
25 init parser module

### LINKED ISSUE #25: Init parser module
## TODO

- [x] Implement `parser`
- [x] Implement `parse_simple_command`
- [x] ~Implement `parse_pipe`~  in #44

### PR DESCRIPTION
**Pull request by rask24**
fix #25

- **Implement parser**
- **Init parse_simple_command**
- **Implement consume_token**
- **Implement parser_utils**
- **Update getter in ast**
- **Fix ast module**
- **Implement parse_simple_command**
- **Refactor parser test**
- **Refactor parse_simple_command**

## Summary by CodeRabbit

- **New Features**
	- Introduced new parsing capabilities with the addition of parser components.
	- Added error handling for syntax errors related to token values.
	- Implemented a new function for handling command parsing in a shell-like environment.
	- Established a grammar specification for command-line interface parsing.

- **Bug Fixes**
	- Enhanced robustness by adding null checks in various functions.

- **Tests**
	- Added new unit tests for parsing functionalities and error handling.
	- Updated existing tests to reflect changes in function names and ensure consistency.

### CODE REVIEW

File: tests/unit/test_parse_simple_commnad.cpp
Comment (alex216) [inline tests/unit/test_parse_simple_commnad.cpp:1]: in 5c9c007 file `tests/unit/test_parse_simple_commnad.cpp` has typo.

File: src/parser/parse_simple_command.c [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	redirect_info.type = (t_redirect_type)token_type;
+	if (!expect_token(cur_token, token_type))
+		return (false);
+	redirect_info.filepath = get_token_value(*cur_token);
+	if (!expect_token(cur_token, TOKEN_WORD))
+		return (false);
+	push_redirect_info(ast, &redirect_info);
+	return (true);
+}
+
```
Comment (alex216) [inline src/parser/parse_simple_command.c:59]:
```
//  simple_command   : cmd_prefix cmd_word cmd_suffix
//                   | cmd_prefix cmd_word
//                   | cmd_prefix
//                   |            cmd_word cmd_suffix
//                   |            cmd_word
```
suggestion to add comment generative grammar

**Review by alex216 — COMMENTED** [review]
please check the comments @rask24

File: src/parser/parse_simple_command.c (hunk 1 above)
Comment (rask24) [reply src/parser/parse_simple_command.c:59]: Good idea 👍🏼 
I've added the current whole grammar to docs/grammar.md. @alex216

**Review by alex216 — APPROVED** [review]
OK @rask24

**Pull request merged**